### PR TITLE
update several deps

### LINKED ISSUE #12773: update stale deps
## Development Task

Replace signal crate with signal-hook as signal introduces several outdated deps.

### PR DESCRIPTION
**Pull request by BusyJay**
### What is changed and how it works?

Issue Number: Close #12773.

What's Changed:

```commit-message
replace signal with signal-hook and update other deps by depbot.
```

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by sticnarf — APPROVED** [review]
LGTM

**Review by sticnarf — CHANGES_REQUESTED** [review]
Oops, tidb_query needs to be fixed to adapt ordered_float changes

**Review by sticnarf — APPROVED** [review]

File: components/external_storage/export/src/bin/tikv-cloud-storage.rs [hunk 1]
```diff
@@ -33,16 +33,19 @@ fn main() {
...
+    };
     use slog_global::info;
 
     pub fn for_signal() {
-        let trap = Trap::trap(&[SIGTERM, SIGINT, SIGHUP, SIGUSR1, SIGUSR2]);
-        for sig in trap {
-            match sig {
-                SIGUSR1 | SIGTERM | SIGINT | SIGHUP => {
-                    info!("receive signal {}, stopping server...", sig as c_int);
+        let mut signals = Signals::new(&[SIGTERM, SIGINT, SIGHUP]).unwrap();
```
[the remark this answers is not in the document]
Comment (BusyJay) [reply components/external_storage/export/src/bin/tikv-cloud-storage.rs:44]: They should not be used at all.

**Review by zhangjinpeng1987 — APPROVED** [review]
LGTM

**Pull request merged**